### DISCUSSION TITLE
Using Mixing Option with Ongoing Stream

### DISCUSSION THREAD

**Discussion by sufyanshafiq** [savonet/liquidsoap#1807] [Q&A] [answered]
I am currently working on creating streams and playing different jingles/ads between ongoing streams. I want to do it after every specific time. Using the option given in complete analysis section (https://www.liquidsoap.info/doc-dev/complete_case.html)
```
radio = add([radio, switch([({0m0s},clock)])])
```
And it working absolutely fine as desired.

I'm using this inside Azuracast, Version of Liquidsoap is 1.4.4 on Ubuntu 20.04 LTS.

**Now here a requirement raised.** 
Is there any possibility while mixing the ongoing stream's volume goes lower down and the jingle/add will be audible and prominent? How to achieve this?

Appreciate the help. Thanks.

**Comment by smimram** [comment]
By default `add` says half of the volume should go to each source (because you have two of them). You can change the relative proeminence with the `weight` parameter. For instance,
```
radio = add(weights=[1,3], [radio, switch([({0m0s},clock)])])
```
says that the `clock` source should be 3 times louder than the `radio` one, when both are available.

**Reply by sufyanshafiq** [reply]
**Hi smimram,**
Thanks for the response. I tried the same option but this makes not difference. Then I tried using it as float just the way explained in the book as below:

> **_Weights_**. The add operator also offers the possibility of weighting the sources, i.e. specifying their relative volume: if we want to hear the microphone twice as loud as the bed, we should give the microphone twice the weight of the bed. The weight of each source can be specified in a list passed in the weights arguments. For instance:
> 
> radio = add(weights=[2., 1.], [mic, bed])

This give the error below:
Error 5: this value has type
  [float]
but it should be a subtype of
  [int]

As per book below I tried apmplify as well:

> assigns the weight 2. to mic and 1. to bed. This is equivalent to amplifying each of the sources with the corresponding factor, i.e.
> radio = add([amplify(2., mic), amplify(1., bed)])
> but more efficient and natural.

radio = add([amplify(1., radio), amplify(2., switch([({0s},playlist_announcements)]))])

This is not giving any error but the result is same i.e. no difference in sound. I would really appreciate your help in this regard. Thanks.

**Reply by sufyanshafiq** [reply]
Is it OK to raise it as issue?

**Reply by smimram** [reply]
For some reason, before version 2.0, Liquidsoap was expecting integer (as opposed to float) weights. You should thus write
```
radio = add(weights=[2, 1], [mic, bed])
```
with those versions.

The `amplify` technique should work. Are you sure you don't hear a difference? Can you try with large coefficients such as 10 or more?

**Reply by sufyanshafiq** [reply]
Yes, giving a larger coefficients I am feeling the difference. But it is keeping the volume throughout as given weights

Isn't it possible that it only shifts volume when playing the announcements giving the effect as songs being played in background with low volume till announcements completed then comes back to normal volume?

**Comment by sufyanshafiq** [comment, marked as answer]
To achieve the scenario I used:

> **Fallback**. The first way of combining sources is through the fallback operator, which takes
as argument a list of sources, and plays the first one which is available, i.e. can produce some
stream.

Below code that helped me:

`mic = blank.strip(max_blank=2., min_noise=.1, threshold=-20., mic)`
`music = playlist("~/Music")`
`radio = fallback(track_sensitive=false, [mic, music])`
